tests/unit/nupic/research/temporal_memory_test.py

```diff
@@ -578,6 +585,73 @@ def testMapCellsToColumns(self):
     self.assertEqual(columnsForCells[99], set([399]))
 
 
+  def testWrite(self):
+    tm1 = TemporalMemory(
+      columnDimensions=[100],
+      cellsPerColumn=4,
+      activationThreshold=7,
+      initialPermanence=0.37,
+      connectedPermanence=0.58,
+      minThreshold=4,
+      maxNewSynapseCount=18,
+      permanenceIncrement=0.23,
+      permanenceDecrement=0.08,
+      seed=91
+    )
+
+    # Run some data through before serializing
+    self.patternMachine = PatternMachine(100, 4)
+    self.sequenceMachine = SequenceMachine(self.patternMachine)
+    sequence = self.sequenceMachine.generateFromNumbers(range(5))
+    for _ in range(3):
+      for pattern in sequence:
+        tm1.compute(pattern)
+
+    proto1 = TemporalMemoryProto_capnp.TemporalMemoryProto.new_message()
+    tm1.write(proto1)
+
+    # Write the proto to a temp file and read it back into a new proto
+    with tempfile.TemporaryFile() as f:
+      proto1.write(f)
+      f.seek(0)
+      proto2 = TemporalMemoryProto_capnp.TemporalMemoryProto.read(f)
+
+    # Load the deserialized proto
+    tm2 = TemporalMemory.read(proto2)
+
+    # Check that the two temporal memory objects have the same attributes
+    self.assertEqual(tm1.columnDimensions, tm2.columnDimensions)
+    self.assertEqual(tm1.cellsPerColumn, tm2.cellsPerColumn)
+    self.assertEqual(tm1.activationThreshold, tm2.activationThreshold)
+    self.assertAlmostEqual(tm1.initialPermanence, tm2.initialPermanence)
+    self.assertAlmostEqual(tm1.connectedPermanence, tm2.connectedPermanence)
+    self.assertEqual(tm1.minThreshold, tm2.minThreshold)
+    self.assertEqual(tm1.maxNewSynapseCount, tm2.maxNewSynapseCount)
+    self.assertAlmostEqual(tm1.permanenceIncrement, tm2.permanenceIncrement)
+    self.assertAlmostEqual(tm1.permanenceDecrement, tm2.permanenceDecrement)
+
+    self.assertEqual(tm1.connections, tm2.connections)
+
+    self.assertEqual(tm1.activeCells, tm2.activeCells)
+    self.assertEqual(tm1.predictiveCells, tm2.predictiveCells)
+    self.assertEqual(tm1.winnerCells, tm2.winnerCells)
+
+    # Run a couple records through after deserializing and check results match
+    tm1.compute(self.patternMachine.get(0))
+    tm2.compute(self.patternMachine.get(0))
+    self.assertEqual(tm1.activeCells, tm2.activeCells)
```

Comment:
`assertEqual` does the same thing as `assertSetEqual` IF they are sets but the later is good practice to enforce that they are sets if that is the API